Docs: Use different color for text in landing page section & Limit search width

danielchalmers opened the pull request:
<!--
MAKE SURE TO READ THE CONTRIBUTING GUIDE BEFORE CREATING A PR
https://github.com/MudBlazor/MudBlazor/blob/dev/CONTRIBUTING.md

Testing sections can be removed for documentation changes
-->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Keep the title short and descriptive, as it will be used as a commit message -->

## Description
<!-- Describe your changes in detail any why -->
<!-- Note any issues that are resolved by this PR -->
<!-- e.g. resolves #1337 or fixes #9310 -->

Fixes regression from #8515 due to the wrong palette being used originally for the text.
Used a gray instead of text-secondary.

Now matches other sections of the landing page.

## How Has This Been Tested?
<!-- All PRs should implement unit tests if possible -->
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | none -->

visually 

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation (fix or improvement to the website or code docs)

<!-- If you made any visual changes, provide screenshots of before/after. If it has moving parts, please provide high quality gif, webm, or mp4 -->

v6

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/596c3d46-dfe1-4bd2-8423-166bc0ac700c)


v7 Before

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/6c955e7e-7700-49f1-9983-5b291d6fc584)


v7 After

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/0884b48d-2943-445c-b202-375536dc97d4)


v7 After Light

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/dddd9843-d5bb-44b8-9ea4-6a92255af80d)


## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] The PR is submitted to the correct branch (`dev`).
- [X] My code follows the code style of this project.
- [ ] I've added relevant tests.

danielchalmers committed:
Docs: Fix landing page text using wrong palette

Garderoben reviewed (changes requested):
Its better yes but why change it from the current site? it still does not match.

danielchalmers commented:
> Its better yes but why change it from the current site? it still does not match.

@Garderoben Because the search bar uses that grey palette in the docs. Adding the search bar to the landing page used the default grey, so I copied it from the docs theme, not realizing that the text used the grey color instead of a text color.

danielchalmers commented:
@Garderoben
If you want I could at some point merge the landing page and doc palettes and unify them further in the backend.
Though really it's the appbar that's the problem. If we could isolate the styles to that it would solve it.

Garderoben commented:
> > Its better yes but why change it from the current site? it still does not match.
> 
> @Garderoben Because the search bar uses that grey palette in the docs. Adding the search bar to the landing page used the default grey and thus looked totally out of place, so I copied it from the docs theme, not realizing that the text used the grey color instead of a text color.
> 
> ![image](https://private-user-images.githubusercontent.com/7112040/320635830-3b9833bd-33dd-40ed-b314-b7ada1876dd2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxMjY0NDEsIm5iZiI6MTcxMzEyNjE0MSwicGF0aCI6Ii83MTEyMDQwLzMyMDYzNTgzMC0zYjk4MzNiZC0zM2RkLTQwZWQtYjMxNC1iN2FkYTE4NzZkZDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTRUMjAyMjIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmJjZmQ4MDFlNjJkYjNlMjU0NTExMjlkYmU3ZWU1NTIxZDRlYjcwYmQzOWEwODcwYWUwODhiOGNmMTZhOWQxNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NHpfirQ0trGIdLkbeDnSQ3CUJrT-GosXP5kO8_l_BVQ)



> @Garderoben If you want I could at some point merge the landing page and doc palettes and unify them further in the backend to fix issues like this and the text color in the search panel
> 
> ![image](https://private-user-images.githubusercontent.com/7112040/322170820-906de23c-aaed-4313-bdfc-92698c445ec8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxMjY0NDEsIm5iZiI6MTcxMzEyNjE0MSwicGF0aCI6Ii83MTEyMDQwLzMyMjE3MDgyMC05MDZkZTIzYy1hYWVkLTQzMTMtYmRmYy05MjY5OGM0NDVlYzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTRUMjAyMjIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjM3OWRhZTc1ODBmMzNiMjljOTBhNjk4YmQxZWE4ZTA4YzlmZTc3NjY0Mjg2YWYwNDU3ODQ4YmE3NmM4Y2VlMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6UKdjZ9BFdGOHLR6uAc6O2UY9a52lU2mAogDI11eNA4)
> 
> The fact that DocsLayout and LandingLayout are separate is slowing me from adding a search hotkey and other features.
> 
> We can keep the nice white/dark contrasty text for the actual landing section and other details like that by scoping the CSS

Yeah i did not design the landing page to have this search bar in the first place, for me it breaks the design of what i did. It just does not look as good.

I don't like the search bar on the landing page but i know @henon probably loves it, we can keep it but i really want the rest of the colors to be the same as before. Docs and Landing page is separated for a reason.

Also the search bar looks bad on bigger monitors, i needs to have the max width set same as on the docs page. (360px)
![image](https://github.com/MudBlazor/MudBlazor/assets/10367109/90ca9a63-cf8e-477a-9f6d-a0b5ecdbd949)

henon commented:
@Garderoben Actually a search button would suffice on the landing page for users like me who visit the site just to look stuff up. Would that be a good compromise?

danielchalmers commented:
> a search button would suffice on the landing page for users like me who visit the site just to look stuff up. Would that be a good compromise?

@henon It was originally a search button that opened a dialog. See [live version](https://mudblazor.com/). Added one extra step to getting to components, but now that I fixed the auto-focus it's not as bad. I prefer search bar but doesn't bother me to revert it.

If you two wanna let me know your immediate goals for the docs (and this PR) I'll probably work on it today or tomorrow.

henon commented:
I just checked, both Vuetify and MUI have a search button on their landing page. I'd say let's keep just the button, so it won't break Garderoben's carefully crafted design ;)

danielchalmers commented:
> Vuetify and MUI

@henon They have a keyboard shortcut though. That's why I was hoping for #7268.

But again, with the auto-focus being fixed it's the same number of clicks anyway, so not really a problem to revert it. Also renders this PR obsolete.

henon commented:
So you got the flaky focus to work consistently?

henon commented:
> That's why I was hoping for #7268.

If the author doesn't respond we can add Adornment RFs on our own.

danielchalmers commented:
> So you got the flaky focus to work consistently?

@henon Seems to work for me on both dev.mudblazor and debug. desktop and mobile. I think it was me plus the person who worked on the focus styles. There is one quirk in which it doesn't trigger the default list but I think that's a core issue with autocomplete and I plan to look into that. Let me know if you run into any problems with it.

Garderoben commented:
> @Garderoben Actually a search button would suffice on the landing page for users like me who visit the site just to look stuff up. Would that be a good compromise?



> > a search button would suffice on the landing page for users like me who visit the site just to look stuff up. Would that be a good compromise?
> 
> @henon It was originally a search button that opened a dialog. See [live version](https://mudblazor.com/). Added one extra step to getting to components, but now that I fixed the auto-focus it's not as bad. I prefer search bar but doesn't bother me to revert it.
> 
> If you two wanna let me know your immediate goals for the docs (and this PR) I'll probably work on it today or tomorrow.

Sorry maybe i was not super clear, with this PR's color changes i'm happy with how it looks so we could keep the new searchbar even though i would prefer the old button but as you say the textfield needs to be auto focused when clicked i notice that its broken now?

What i meant was i would like no further color changes after this.

danielchalmers committed:
add max width from docs layout

danielchalmers commented:
@Garderoben Just pushed a commit to make the search bar the same size as the docs one. Should be ready to merge if the rest of the page looks good.

Just so we're on the same page, there are some subtle differences between the two layouts:

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/b749187e-6ae3-4aa4-9d1a-34d8eb84c2c5)

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/682aacc6-c3b4-4a18-bbae-88398e6bbbf9)

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/bd3058b3-a0cc-4e8f-a457-390272f5a0e2)

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/b3902f0e-f38a-46c5-8d72-76bbb55a538d)

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/8105deaf-63b6-4ae0-aa25-b4180dad0b43)

![image](https://github.com/MudBlazor/MudBlazor/assets/7112040/0183a1d4-91ad-40f4-ab9b-6f0eba46a6d8)


Doesn't matter to me either way but I'm available if anyone decides they want the backends unified to solve those issues while keeping the actual landing page content visually the same.

jperson2000 commented:
This looks like a docs-only change.  You can add this to the Types of Changes section to tag it with `docs`

```
- [X] Documentation (fix or improvement to the website or code docs)
```
